### DISCUSSION TITLE
Great Project, need more documentation /guides

### DISCUSSION THREAD

**Discussion by msreekm** [UnlyEd/next-right-now#134] [General] [answered]
I was navigating the examples and documentation. one suggestion is to document  how does it all work  details of what is in _app, document etc, i couldn't find them. the other question that popped up to my mind is whether the logging,analytics etc is pluggable , if we are already using xyz for logging, monitoring etc ,how do we use them?

**Comment by Vadorequest** [comment, marked as answer]
@msreekm Thanks for your feedback, trying to summarize it into actionable steps, it'd be:

1. Document how the "bootstrap" of the app is done (_app)
    - Where would you rather see this being documented? I'm thinking at https://unlyed.github.io/next-right-now/reference/, what do you think? It could also be added to the [FAQ](https://unlyed.github.io/next-right-now/faq) IMO
2. Regarding "About using a different logging/analytics tools, how can it be integrated?", I think it should be part of the FAQ 
    - To answer your question, if you use a different system, then you should read how to get rid of what's built-in (assuming you won't use it), and then implement your system on your own, NRN can't help much if you're not using something that's built-in, you'll need to figure that part out mostly by yourself. The [guides](https://unlyed.github.io/next-right-now/guides/analytics/remove-amplitude) section explain how to remove what's built-in.

**Reply by msreekm** [reply]
one related quesiton on the SSR example .in getserversideprops ,you are querying for customer, assume that it is a list of customers and you get first 10 customers on server side. now on the client side on initial render it will show 1-10 customers, how do you now pull next 10 customers (pagination) ? if you use apollo useQuery for example, useQuery returns a fetch that is used for next query, in this case the first query was done on server side . how does it continue the rest of client side?

**Reply by Vadorequest** [reply]
The `Customer` entity is very particular and related to the MST design as explained [here](https://unlyed.github.io/next-right-now/concepts/tenancy.html#multiple-single-tenancy), and you won't be querying multiple customers anyway.

But I understand the underlying question, which is about pagination and how to fetch data from either the server or the client side. Next allows both, and you can totally build a pagination system where you run your queries either on the server side, or the client side, or both (server side request first, then client side request)

It also depends on if you want to use static pages (SSG), in such case you'd prefetch all data during build time and build your pages at once.

Eventually, this question is about how Next.js works and I advise you rather ask this on their Discussion, if you want to take it deeper. https://github.com/vercel/next.js/discussions

**Reply by Vadorequest** [reply]
The documentation has been updated to better explain the above-mentioned topics.
See https://unlyed.github.io/next-right-now/concepts/app-bootstrap.html
